src/node/internal/internal_fs_sync.ts

```diff
@@ -0,0 +1,1066 @@
+/* eslint-disable @typescript-eslint/no-unused-vars,@typescript-eslint/no-unnecessary-condition */
+
+import {
+  kMaxUserId,
+  stringToFlags,
+  getValidatedFd,
+  validateBufferArray,
+  validateStringAfterArrayBufferView,
+  normalizePath,
+  Stats,
+  kBadge,
+  type FilePath,
+  validatePosition,
+} from 'node-internal:internal_fs_utils';
+import {
+  validateInteger,
+  parseFileMode,
+  validateBoolean,
+  validateObject,
+  validateOneOf,
+  validateEncoding,
+  validateUint32,
+} from 'node-internal:validators';
+import {
+  ERR_INVALID_ARG_TYPE,
+  ERR_INVALID_ARG_VALUE,
+} from 'node-internal:internal_errors';
+import { isArrayBufferView } from 'node-internal:internal_types';
+import {
+  F_OK,
+  X_OK,
+  W_OK,
+  O_WRONLY,
+  O_RDWR,
+  O_APPEND,
+  O_EXCL,
+  COPYFILE_EXCL,
+  COPYFILE_FICLONE_FORCE,
+} from 'node-internal:internal_fs_constants';
+import { type Dir, Dirent } from 'node-internal:internal_fs';
+import { default as cffs } from 'cloudflare-internal:filesystem';
+
+import { Buffer } from 'node-internal:internal_buffer';
+throw new ERR_();
+function accessSyncImpl(
+  path: FilePath,
+  mode: number = F_OK,
+  followSymlinks: boolean
+): void {
+  validateUint32(mode, 'mode');
+
+  // If the X_OK flag is set we will always throw because we don't
+  // support executable files.
+  if (mode & X_OK) {
+    throw new Error('access denied'); // not executable.
```

Comment:
It would be great to see the error system developed here, as it's critical to most FS workflows. Left some notes on the C++ side as well but completely understood if it's a follow-on.